### PR TITLE
MAINT: cloned manifold_learner

### LINKED ISSUE #22: MAINT: Make copy of `manifold_learner` before using it, if passed to the constructor.
Currently no copy is made, and the passed learner is modified in-place.
https://github.com/zoj613/pyloras/blob/d39c5b583513df251059aee8b36f5022a7e82bab/pyloras/_loras.py#L129-L131
https://github.com/zoj613/pyloras/blob/d39c5b583513df251059aee8b36f5022a7e82bab/pyloras/_loras.py#L134-L139

A copy should be made using sklearn's `clone` function so as to preserve the passed object.

### PR DESCRIPTION
**Pull request by vortex-17**
Hi, this is the pull request i have made according to the problem (https://github.com/zoj613/pyloras/issues/22)
I hope this solves it.

I have assumed that self.manifold_learner_ is the copied version. Let me know if i need to make any changes.
Thank you !

### CODE REVIEW

File: pyloras/_loras.py [hunk 1]
```diff
@@ -128,7 +129,7 @@ def _initialize_params(self, X, y, rng):
 
         if self.manifold_learner:
             self._check_2d_manifold_learner()
-            self.manifold_learner_ = self.manifold_learner
```
Comment (zoj613) [inline pyloras/_loras.py:131]: This looks good @vortex-17 , thank you for taking the time to contribute. I think we could now just return the cloned manifold_learner using the `_check_2d_manifold_learner` method to make things look a little cleaner and have a single line like:
```python
self.manifold_learner_ = self._check_2d_manifold_learner()
```

**Comment by zoj613** [comment]
@vortex-17 It seems this needs a rebase with main branch to resolve the conflict.

**Comment by vortex-17** [comment]
> @vortex-17 It seems this needs a rebase with main branch to resolve the conflict.

Should i create another pull request with git rebase and changes (self.manifold_learner_ = self._check_2d_manifold_learner()) ?

**Comment by zoj613** [comment]
> > @vortex-17 It seems this needs a rebase with main branch to resolve the conflict.
> 
> Should i create another pull request with git rebase and changes (self.manifold_learner_ = self._check_2d_manifold_learner()) ?

You do not need to create another, just update your changes on this same PR.

**Comment by vortex-17** [comment]
> > > @vortex-17 It seems this needs a rebase with main branch to resolve the conflict.
> > 
> > 
> > Should i create another pull request with git rebase and changes (self.manifold_learner_ = self._check_2d_manifold_learner()) ?
> 
> You do not need to create another, just update your changes on this same PR.

I have made the changes in a new branch "cloning" which has the updated code. Do i send a new PR with that branch ? 
I apologise if this is annoying as i am new to open source.

**Comment by zoj613** [comment]
> > > > @vortex-17 It seems this needs a rebase with main branch to resolve the conflict.
> > > 
> > > 
> > > Should i create another pull request with git rebase and changes (self.manifold_learner_ = self._check_2d_manifold_learner()) ?
> > 
> > 
> > You do not need to create another, just update your changes on this same PR.
> 
> I have made the changes in a new branch "cloning" which has the updated code. Do i send a new PR with that branch ?
> I apologise if this is annoying as i am new to open source.

No worries, using git effectively when contributing to projects takes some getting used to. You can submit the updated changes as a new PR.

Check out this article for future references on how to get started with using rebasing to resolve merge conflict: https://phoenixnap.com/kb/how-to-resolve-merge-conflicts-in-git

In most cases, you dont even need to rebase when there is no merge-conflict, you just add a new commit on top of the existing one in the same branch and PR. See here for example: https://github.com/zoj613/pyloras/pull/20/commits

**Comment by vortex-17** [comment]
Heyy can you check now ? I have tried to rebase and fix merge issues. 
Or i will PR from the other branch.
Thanks a lot

**Comment by zoj613** [comment]
> Heyy can you check now ? I have tried to rebase and fix merge issues.
> Or i will PR from the other branch.
> Thanks a lot

Unfortunately, the rebase did not work as expected. You added the base branch's commits on top of yours and then duplicated your commit once more on top of the existing commits.
For now, lets try a new PR from another branch. This way you wont get frustrated by learning how to rebase correctly on the fly. You can learn how to fix these issues by practicing on your local machine using this branch.

**Pull request closed without merging**